### PR TITLE
Add github action to test diagnostics

### PR DESCRIPTION
**Pull request by jmbockhorst**
This should give a "Run Manually" button on the actions tab. We could also set it up on a 24 hour schedule.

### CODE REVIEW

**Comment by Razzeee** [comment]
The reason to not have this on the normal build is due to it taking too long?

**Comment by jmbockhorst** [comment]
Yes. I'm running it now on my personal repo and it's been going for over 2 hours. It might be froze up because I can't see the logs.
https://github.com/jmbockhorst/elm-language-server/runs/1493270378?check_suite_focus=true

We can see, but I would suspect that even if it ran smooth it would take at least 30 minutes to an hour, or possibly more. Didn't think we'd want that on every push, but we can consider.

**Comment by Razzeee** [comment]
Yeah, was just wondering what too long is in this case :)

**Comment by jmbockhorst** [comment]
Actually, it looks like it just keeps getting stuck. Otherwise it should complete in 11-12 minutes.

**Comment by jmbockhorst** [comment]
This is ready to go now. Currently it doesn’t fail when a library fails so that we can test them. We can enable that once they are all passing or we have a known failures list.  Unfortunately it looks like there are about 150 failures, which is way more than I saw last week. I will be looking at the workspaces invididually over the next week trying to find the issues.

File: test/diagnostics.ts [hunk 1]
```diff
@@ -25,8 +19,8 @@ container.register("Connection", {
       warn: (): void => {
         // console.log(a);
       },
-      error: (): void => {
-        // console.log(a);
+      error: (a: string): void => {
+        console.log(a);
```
Comment (Razzeee) [inline test/diagnostics.ts:23]: Do you want to keep it like this?

File: test/diagnostics.ts [hunk 2]
```diff
@@ -50,61 +45,59 @@ async function initParser(): Promise<void> {
...
     console.log(`${diagnostics.length} diagnostics found.`);
 
     diagnostics.forEach((diagnostic) => {
-      console.log(
-        `${path.basename((<any>diagnostic.data).uri)}: ${diagnostic.message}`,
-      );
+      console.log(`${path.basename(diagnostic.uri)}: ${diagnostic.message}`);
     });
 
+    console.log();
```
Comment (Razzeee) [inline test/diagnostics.ts:84]: Empty log?

File: test/diagnostics.ts [hunk 3]
```diff
@@ -160,26 +150,50 @@ const filteredLibs = libsToParse
     (lib) =>
       !lib.startsWith("elm/") &&
       !lib.startsWith("elm-explorations/") &&
-      !parsingFailures.includes(lib) &&
-      !compilerFailures.includes(lib) &&
+      !unknownFailures.includes(lib) &&
+      // !parsingFailures.includes(lib) &&
+      // !compilerFailures.includes(lib) &&
```
Comment (Razzeee) [inline test/diagnostics.ts:155]: should these still be commented out? should we remove the variables?

**Review by Razzeee — CHANGES_REQUESTED** [review]
Some cleanup might be needed

File: test/diagnostics.ts (hunk 3 above)
Comment (jmbockhorst) [reply test/diagnostics.ts:155]: Going to re add them once we figure out which ones are actually parser or compile errors

File: test/diagnostics.ts (hunk 1 above)
Comment (jmbockhorst) [reply test/diagnostics.ts:23]: Yeah I think it’ll help if there are any errors.

File: test/diagnostics.ts (hunk 2 above)
Comment (jmbockhorst) [reply test/diagnostics.ts:84]: Just to add a space.

**Pull request merged**